backend/CircleForms/Services/Request/OsuUserProvider.cs

```diff
@@ -1,27 +1,57 @@
-﻿using System;
+﻿using System.Net;
 using System.Threading.Tasks;
+using AutoMapper;
+using CircleForms.Models;
+using CircleForms.Models.Configurations;
 using CircleForms.Models.OsuContracts;
 using CircleForms.Services.Interfaces;
+using Microsoft.Extensions.Options;
 using RestSharp;
+using RestSharp.Serializers.NewtonsoftJson;
 
 namespace CircleForms.Services.Request;
 
-public class OsuUserProvider : IOsuUserProvider
+public class OsuApiProvider : IOsuApiProvider
 {
-    private const string _apiLink = "https://osu.ppy.sh/api/v2/me";
-    private readonly IRestClient _client;
+    private const string _apiMeLink = "https://osu.ppy.sh/api/v2/me";
+    private const string _apiTokenLink = "https://osu.ppy.sh/oauth/token";
+    private readonly RestClient _getUserClient = new RestClient(_apiMeLink).UseNewtonsoftJson();
```

Comment:
It should probably be one `RestClient` and multiple `RestRequest`s instead of multiple `RestClient`s since `RestClient` is a `HttpClient` wrapper.

Reply:
I dont think there is the way to change api destination in one client, so i believe that's how it's done

Reply:
If I remember correctly, you set *base* address in the client (`https://osu.ppy.sh/` is our case) and the endpoint in the `RestRequest`

Reply:
ok i got it